cooltools/insulation.py

```diff
@@ -70,16 +76,16 @@ def insul_diamond(pixel_query, bins, window=10, ignore_diags=2, norm_by_median=T
     sum_balanced = np.zeros(N)
 
```

Comment:
should probably add that  `is_cooler_balanced` check here ...
I haven't propagated that check to many tools yet (should be everywhere basically) , but here is one example https://github.com/open2c/cooltools/blob/705f5d4e895aa4c16df6c210c7f1987a4754a978/cooltools/expected.py#L1459

for this case we could do some simple exception catching this:
```python
try:
   _ = is_cooler_balanced(clr, clr_weight_name, raise_errors=True)
except Exception as e:
    raise(f"cooler is not balanced or {clr_weight_name} is missing") from e
```

not sure it is useful, but it would be sort of trivial to allow `clr_weight_name=None` - i.e. insulation on unbalanced data as well . Not for v0.5 i think

Reply:
moved this check to the user-facing function instead